docs/manager/graphql-reference/schema.graphql

```diff
@@ -9,6 +9,37 @@ type Queries {
     """The ID of the object"""
     id: ID!
   ): Node
+
+  """Added in 25.5.0."""
+  audit_log_schema: AuditLogSchema
+
+  """Added in 25.5.0."""
+  audit_log_nodes(
+    """
+    Specifies the criteria used to narrow down the query results based on certain conditions.
+    """
+    filter: String
+
+    """Specifies the sorting order of the query result."""
+    order: String
+
+    """Specifies how many items to skip before beginning to return result."""
+    offset: Int
+
+    """If this value is provided, the query will be limited to that value."""
+    before: String
+
+    """Queries the `last` number of results from the query result from last."""
+    after: String
+
+    """
+    Queries the `first` number of results from the query result from first.
+    """
+    first: Int
+
+    """If the given value is provided, the query will start from that value."""
+    last: Int
```

Comment:
Is it possible to make them types and bundle them together instead of listing them separately?

Reply:
Yes, in order to do that, I think we need to define `AsyncPaginatedConnectionV2Field`.

https://github.com/lablup/backend.ai/blob/560953388c41f91596c6095aa60f53edd3c7a56c/src/ai/backend/manager/models/base.py#L1420-L1425

I’ll try making the change.

Reply:
Looking again, it seems that `IterableConnectionField`, the parent class of `AsyncListConnectionField` (a type from graphene), dynamically adds `before`, `after`, `first`, and `last`.

It looks like we need to define and use our own version of `IterableConnectionField`.

Reply:
Isn't this a work in progress? @jopemachine